app/controllers/DisclaimerController.scala

```diff
@@ -36,15 +37,15 @@ class DisclaimerController @Inject()(
                                          navigator: Navigator,
                                          val controllerComponents: MessagesControllerComponents,
                                          disclaimerView: DisclaimerView
-                                       ) extends FrontendBaseController with I18nSupport with Logging with UIAssembler {
+                                       ) extends FrontendBaseController with I18nSupport with Logging {
 
   def onPageLoad: Action[AnyContent] = (identify andThen citizenDetailsCheck andThen getData andThen requireData) {
     implicit request =>
       request.userAnswers.get(SelectTaxYearsToClaimForPage) match {
-        case Some(_) =>
-          val selectedTaxYears = taxYearFromUIAssemblerFromRequest()
-
-          Ok(disclaimerView(selectedTaxYears.contains2022OrAfter, selectedTaxYears.contains2021Or2020))
+        case Some(selectedTaxYears) =>
+            val contains2020Or2021 = selectedTaxYears.intersect(contains2020or2021).nonEmpty
+            val contains2022OrAfter = selectedTaxYears.intersect(contains2022orAfter).nonEmpty
```

Comment:
You could move the logic into the model as a def rather than just defining a list there.